webapp/WEB-INF/conf/core_context.xml

```diff
@@ -50,7 +53,10 @@
     <bean id="siteMessageHandler" class="fr.paris.lutece.portal.service.message.SiteMessageHandler"/>
 
     <!--                      Search Engine                                   -->
-    <bean id="searchEngine" class="fr.paris.lutece.portal.service.search.LuceneSearchEngine" />
+    <bean id="searchEngine" class="fr.paris.lutece.portal.service.search.LuceneSearchEngine">
+        <aop:scoped-proxy proxy-target-class="false" />
```

Comment:
I effectively could not find much documentation on this. In the video by a spring developer linked in the thread you refer to, it's mentioned to just work beginning with spring 3 when in a web application context (which is the case with Lutece). However, I found it necessary to use scoped-proxy for the serialization to work. 

Maybe the fact the spring version used by Lutece is quite ancient does not help.

Note that this is of concern _only_ if you care about session serializability. If you do not, you can just leave the directive out and still benefit from spring facilities for your XPageApplication (most notably dependency injection), and allow for easy overrides of the implementation of apps. 

I converted SearchApp over as a proof of concept and to allow testability. Lutece core with the patch has an app managed by Spring (SearchApp) and an app not managed by spring (SiteMapApp) and both cases are tested. 
Maybe a specific test on serialization could be added. 

> Le 23 juil. 2015 à 17:04, Jon Harper <notifications@github.com> a écrit :
> 
> In webapp/WEB-INF/conf/core_context.xml:
> 
> > @@ -50,7 +53,10 @@
> >      <bean id="siteMessageHandler" class="fr.paris.lutece.portal.service.message.SiteMessageHandler"/>
> >  
> >      <!--                      Search Engine                                   -->
> > -    <bean id="searchEngine" class="fr.paris.lutece.portal.service.search.LuceneSearchEngine" />
> > +    <bean id="searchEngine" class="fr.paris.lutece.portal.service.search.LuceneSearchEngine">
> > +        <aop:scoped-proxy proxy-target-class="false" />
> I can't find any docs on using aop-proxy this way. Other people have done it though: http://stackoverflow.com/questions/3180963/spring-session-scoped-beans-controllers-and-references-to-services-in-terms-o
> Is it totally undocumented ? If so, maybe we should just keep the local variable and the call to getBean on each method call ?
> 
> —
> Reply to this email directly or view it on GitHub.
>